lib/stub_ui/lib/src/ui/text.dart

```diff
@@ -1559,6 +1615,57 @@ class ParagraphBuilder {
     _ops.add(text);
   }
 
+  /// Adds an inline placeholder space to the paragraph.
+  ///
+  /// The paragraph will contain a rectangular space with no text of the dimensions
+  /// specified.
+  ///
+  /// The [width] and [height] parameters specify the size of the placeholder rectangle.
+  ///
+  /// The [alignment] parameter specifies how the placeholder rectangle will be vertically
+  /// aligned with the surrounding text. When [PlaceholderAlignment.baseline],
+  /// [PlaceholderAlignment.aboveBaseline], and [PlaceholderAlignment.belowBaseline]
+  /// alignment modes are used, the baseline needs to be set with the [baseline].
+  /// When using [PlaceholderAlignment.baseline], [baselineOffset] indicates the distance
+  /// of the baseline down from the top of of the rectangle. The default [baselineOffset]
+  /// is the [height].
+  ///
+  /// Examples:
+  /// * For a 30x50 placeholder with the bottom edge aligned with the bottom of the text, use:
+  /// `addPlaceholder(30, 50, PlaceholderAlignment.bottom);`
+  /// * For a 30x50 placeholder that is vertically centered around the text, use:
+  /// `addPlaceholder(30, 50, PlaceholderAlignment.middle);`.
+  /// * For a 30x50 placeholder that sits completely on top of the alphabetic baseline, use:
+  /// `addPlaceholder(30, 50, PlaceholderAlignment.aboveBaseline, baseline: TextBaseline.alphabetic)`.
+  /// * For a 30x50 placeholder with 40 pixels above and 10 pixels below the alphabetic baseline, use:
+  /// `addPlaceholder(30, 50, PlaceholderAlignment.baseline, baseline: TextBaseline.alphabetic, baselineOffset: 40)`.
```

Comment:
(for extra bonus points) I kinda wish these examples had little images embedded showcasing what all this would look like.